### PR TITLE
Expose `ember-template-recast` API on main entry point (e.g. `require('ember-template-lint').recast`)

### PR DESCRIPTION
**Pull request by dcyriller**
This is a second take for https://github.com/ember-template-lint/ember-template-lint/pull/1251.

### CODE REVIEW

File: lib/linter.js [hunk 1]
```diff
@@ -232,7 +232,7 @@ class Linter {
         fileConfig,
       });
 
-      let { code } = transform(currentSource, (env) => rule.getVisitor(env));
+      let { code } = transform(currentSource, () => rule.getVisitor());
```
Comment (Turbo87) [inline lib/linter.js:235]: technically this is a breaking change, now that it had already been released 😞

Comment (dcyriller) [reply lib/linter.js:235]: Should we keep it or should we release this as a bug fix?

Comment (Turbo87) [reply lib/linter.js:235]: I'll let @rwjblue decide on this :)

Comment (rwjblue) [reply lib/linter.js:235]: I'd just leave it alone, and we can remove for next major. Doesn't really hurt anything to pass the argument...

Comment (dcyriller) [reply lib/linter.js]: ya, updated 👍

**Comment by rwjblue** [comment]
Released in https://github.com/ember-template-lint/ember-template-lint/releases/tag/v2.5.1

**Pull request merged**